go/vt/vterrors/code.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vterrors
+
+import (
+	"fmt"
+
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+)
+
+var (
+	VT03001 = errorWithState("VT03001", vtrpcpb.Code_INVALID_ARGUMENT, SyntaxError, "aggregate functions take a single argument '%s'", "The planner only accepts aggregate functions that take a single argument.")
```

Comment:
This is not something for this PR, but is there a usage also for callers to get more structured information from errors? So like here, would it be useful to have for example the specific aggregate function and arguments to more specifically describe the error for the user?

It means we'd likely need explicit error struct types for this, but I don't know if we have then a good way to have that working across the GRPC layer so the structure isn't lost there. Anyway, not something for here but maybe something to consider for future improvements.